pkg/storage/chunk/client/alibaba/oss_object_client.go

```diff
@@ -71,6 +71,23 @@ func NewOssObjectClient(_ context.Context, cfg OssConfig) (client.ObjectClient,
 func (s *OssObjectClient) Stop() {
 }
 
+func (s *OssObjectClient) ObjectExists(ctx context.Context, objectKey string) (bool, error) {
+	var options []oss.Option
+	err := instrument.CollectedRequest(ctx, "OSS.ObjectExists", ossRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		_, requestErr := s.defaultBucket.GetObjectMeta(objectKey, options...)
+		return requestErr
+	})
+	if err == nil {
+		return true, nil
+	}
+
+	if s.IsObjectNotFoundErr(err) {
```

Comment:
We never make a distinction between if the object is not found or some other error. Would it make sense to just 
```go
if err != nil {
  return false, err
}

return true, nil
```

Reply:
hah agreed. I pushed a commit removing that extra check, lmk what you think.